### DISCUSSION TITLE
Bug: PR component doesn't render some of the time

### DISCUSSION THREAD

**Discussion by charliemic** [community/community#62998] [Pull Requests] [answered]
### Select Topic Area

Bug

### Body

We’re experiencing an issue where for some PRs the merge component doesn’t render on GitHub.com (see screenshot below).

![image](https://github.com/community/community/assets/30408172/e30af85b-8a90-44f6-bfa4-efbd487c1965)

It appears for these PRs, a request to the `/merging` endpoint returns 502 Gateway Timeout (but with HTML in the payload showing a timeout error page with a unicorn visual). In successful PRs, this returns the HTML to render that component.

We've not been able to identify which PRs will exhibit this behaviour. Closing and re-opening a PR exhibiting this behaviour often resolves it.

**Comment by charliemic** [comment]
Pretty sure it's a server side issue.

Client side issues ruled out:

- Browser (tested different browsers)
- Cookies (incognito & cleared cookies)
- Extensions (tested no extensions)
- Network (tested multiple different networks and VPNs)

Two other pieces of evidence that suggest server-side:
- The 502 response from the server
- Experienced by multiple users on the same PRs with different client setups

**Reply by mehulrajdev** [reply]
Refresh it again or use the other incognito tab

mark this as the answer

**Reply by charliemic** [reply]
It's definitely not client side. The issue persists consistently for the same PRs regardless of cookies or client state. 

A 502 response from the server co-inciding with 16s+ response time from a particular endpoint indicates some server-side issue, most likely poor optimisation on a database query (or queries).

**Comment by charliemic** [comment, marked as answer]
Received a response from Github support that this was a known bug that was fixed in a release around 5th August
